src/styles/main.scss

```diff
@@ -1,3 +1,79 @@
+$paddingCalendar: 10px;
+$gapFeature: 1px;
+$sideOfSquare: 100px;
+$widthOfCalendar: (
+  7 * $sideOfSquare) + (6 * $gapFeature) + (2 * $paddingCalendar
+);
+$marginFromLeftSide: (
+  "mon": 0,
+  "tue": 1,
+  "wed": 2,
+  "thu": 3,
+  "fri": 4,
+  "sat": 5,
+  "sun": 6
+);
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.centered {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $widthOfCalendar;
+  padding: 10px;
+
+  @for $item from 1 through 31 {
+    &__day:nth-child(#{$item}) {
+      position: relative;
+      bottom: 0;
+      display: none;
+      transition: all 0.5s;
+      transition-timing-function: cubic-bezier(0, 0.48, 0.31, 0.93);
+      box-sizing: border-box;
+      width: $sideOfSquare;
+      height: $sideOfSquare;
+      border: 1px solid black;
+      background-color: #eee;
+      font-size: 30px;
+
+      &::before {
+        content: "#{$item}";
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+      }
+    }
+  }
+
+  &__day:hover {
+    bottom: 20px;
+    cursor: pointer;
+    background-color: pink;
+  }
+
+  @each $item, $left in $marginFromLeftSide {
+    &--start-day--#{$item} :first-child {
+      margin-left: calc(#{$left} * 101px);
+    }
+  }
+
+  @for $item from 28 through 31 {
+    &--month-length--#{$item} {
+      & :nth-child(-n + #{$item}) {
+        display: block;
```

Comment:
if you do the same vice-versa (display: none for items you wanna hide and display: block by default) less amount of CSS code will be generated from SCSS